### DISCUSSION TITLE
Simplest way to determine device based cost over half hourly intervals

### DISCUSSION THREAD

**Discussion by willenic** [BottlecapDave/HomeAssistant-OctopusEnergy#443] [Q&A] [answered]
Hi Folks,

What would be the simplest way to apply my half hourly agile tariff to a sensor showing device energy consumption. I have a utility helper showing consumption of energy here -->

<img width="195" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/144794340/afa9d4a9-69ab-4ab9-bfb9-4d8ac01aa941">

However the utility helper doesn't seem to support the current_rate. I've been going round in circles on youtube and if I see that bl**dy ad for AI based forex trading one more time, I'll end up driving to Henley to search for the bloke in his triathlon training man cave 😄.

Ultimately I want a live dashboard showing breakdown consumption and cost by half hour for Total, Heating, Hot Water, Car, Cooking. The energy dashboard only goes to hours so doesn't really help me see how well I've optimised my automations to make use of the Agile tariff.

TIA

Will

**Comment by BottlecapDave** [comment, marked as answer]
While I haven't tried this, you should be able to create a template sensor which multiples the energy sensor with the current price sensor. You could then use a utility meter which monitors this template sensor and resets every day to give you a daily cost. The utility meter will need delta values set to true as the template sensor will go up and down as energy is used.

**Reply by willenic** [reply]
Thanks again @BottlecapDave , I kind of figured that template sensor would be the way to go but also thought there might be a simpler way to do it from the UI in HA. Coffe coming your way, looking forward to the new release.
